tap_workday_raas/discover.py

```diff
@@ -8,39 +8,94 @@
 
 LOGGER = singer.get_logger()
 
-def _type_to_schema(elem_type):
+def _element_to_schema(element):
+    elem_type = element.attrib['type'].split(':')[1]
+    is_nullable = element.attrib.get('minOccurs') == '0'
+    # NB: The only actual value I've seen for maxOccurs is "unbounded". There may be other possible values
+    is_list = element.attrib.get("maxOccurs") == 'unbounded'
+
+    schema = {}
+
     if elem_type in ('date', 'dateTime'):
-        return {
-            'type': ['string', 'null'],
+        schema = {
+            'type': ['string'],
             'format': 'date-time'
         }
     elif elem_type == 'decimal':
         #TODO What else do we need for decimal types?
-        return {
-            'type': ['number', 'null']
+        schema = {
+            'type': ['number'],
+            'format': 'decimal'
         }
     else:
-        return {'type': [elem_type, 'null']}
+        schema = {'type': [elem_type]}
+
+    if is_nullable:
+        schema['type'].append('null')
+
+    if is_list:
+        schema = {
+            'type': 'array',
+            'items': schema
+        }
+
+    return schema
+
+# TODO: Test this logic
```

Comment:
Has this been to-did? Are there any cases here where we'd want to hard fail if we see something we don't know about?

Reply:
This is to-done.